src/iterative_ensemble_smoother/_iterative_ensemble_smoother.py

```diff
@@ -123,8 +124,24 @@ def fit(
         if noise is None:
             noise = rng.standard_normal(size=(num_obs, ensemble_size))
 
-        E = make_E(observation_errors, noise)
+        # Columns of E should be sampled from N(0,Cdd) and centered, Evensen 2019
+        if R is not None:
+            # Assuming R and observation_errors has relation
+            # R = Cdd / np.sqrt(observation_errors) from function _create_errors()
+            Cdd = R * observation_errors**2
+        else:
+            Cdd = np.diag(observation_errors**2)
+        eval_cdd, evec_cdd = np.linalg.eigh(Cdd)
+        Cdd_sqrt = evec_cdd * np.sqrt(eval_cdd) @ evec_cdd.T
+        E = np.array([Cdd_sqrt @ z for z in noise.T]).reshape(ensemble_size, num_obs).T
```

Comment:
could perhaps just be written `E = Cdd_sqrt @ noise` ?

Reply:
I think something like this might work:

```python
E = np.linalg.cholesky(Cdd) @ noise
```

Reply:
Thanks, yes this is much better.